### PR TITLE
updated banner

### PR DESCRIPTION
**Pull request by shawndwells**
The banner text incorrectly had spaces around a few dashes. Should resolve https://github.com/ComplianceAsCode/content/issues/4387

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/rule.yml [hunk 1]
```diff
@@ -11,26 +11,19 @@ description: |-
...
+    <br />
+    By using this IS (which includes any device attached to this IS), you consent to the following conditions:<br />
+    <br />
+    -The USG routinely intercepts and monitors communications on this IS for purposes including, but not limited to, penetration testing, COMSEC monitoring, network operations and defense, personnel misconduct (PM), law enforcement (LE), and counterintelligence (CI) investigations.<br />
+    <br />
+    -At any time, the USG may inspect and seize data stored on this IS.<br />
+    <br />
+    -Communications using, or data stored on, this IS are not private, are subject to routine monitoring, interception, and search, and may be disclosed or used for any USG-authorized purpose.<br />
+    <br />
+    -This IS includes security measures (e.g., authentication and access controls) to protect USG interests--not for your personal benefit or privacy.<br />
```
Comment (shawndwells) [inline linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/rule.yml]: This was the part which incorrectly had spaces around the two dashes (old line 26).

**Comment by shawndwells** [comment]
Used https://dodcio.defense.gov/Portals/0/Documents/DoDBanner-9May2008-ocr.pdf as source of correct text.

**Comment by shawndwells** [comment]
Also added a net-new "warning" section reminding users of exact text being needed.

File: linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/rule.yml [hunk 2]
```diff
@@ -79,3 +72,13 @@ ocil: |-
     To check if the system login banner is compliant,
     run the following command:
     <pre>$ cat /etc/issue</pre>
+
+warnings:
+    - general: |-
+        The DoD "Policy on Use of Department of Defense (DoD) Information
+        Systems - Standard Consent Banner and User Agreement" requires
+        an exact text match, including line breaks and spaces, of the
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/rule.yml]: @shawndwells The regular expression used for this rule, defined in https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-banners/login_banner_text.var, allows arbitrary line breaks and spaces. Should https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-banners/login_banner_text.var be changed to require only exact text match?

**Comment by shawndwells** [comment]
On 6/11/19 8:42 AM, Jan Černý wrote:
>
> *@jan-cerny* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In 
> linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/rule.yml 
> <https://github.com/ComplianceAsCode/content/pull/4389#discussion_r292433122>:
>
> > @@ -79,3 +72,13 @@ ocil: |-
>       To check if the system login banner is compliant,
>       run the following command:
>       <pre>$ cat /etc/issue</pre>
> +
> +warnings:
> +    - general: |-
> +        The DoD "Policy on Use of Department of Defense (DoD) Information
> +        Systems - Standard Consent Banner and User Agreement" requires
> +        an exact text match, including line breaks and spaces, of the
>
> @shawndwells <https://github.com/shawndwells> The regular expression 
> used for this rule, defined in 
> https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-banners/login_banner_text.var, 
> allows arbitrary line breaks and spaces. Should 
> https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-banners/login_banner_text.var 
> be changed to require only exact text match?
>

Probably. But that can be done in a future PR.

In the mean time, someone could look at the DISA-provided OVAL and see 
how their regex looks.

**Comment by jan-cerny** [comment]
In Red Hat Enterprise Linux 7 - Ver 2, Rel 3 STIG they thave

``` 
^(\\n|\s)*You\s+are\s+accessing\s+a\s+U\.S\.\s+Government\s+\(USG\)\s+Information\s+System\s+\(IS\)\s+that\s+is\s+provided\s+for\s+USG-authorized\s+use\s+only\.\s*(\\n|\n)+\s*By\s+using\s+this\s+IS\s+\(which\s+includes\s+any\s+device\s+attached\s+to\s+this\s+IS\),\s+you\s+consent\s+to\s+the\s+following\s+conditions\:\s*(\\n|\n)+\s*-The\s+USG\s+routinely\s+intercepts\s+and\s+monitors\s+communications\s+on\s+this\s+IS\s+for\s+purposes\s+including,\s+but\s+not\s+limited\s+to,\s+penetration\s+testing,\s+COMSEC\s+monitoring,\s+network\s+operations\s+and\s+defense,\s+personnel\s+misconduct\s+\(PM\),\s+law\s+enforcement\s+\(LE\),\s+and\s+counterintelligence\s+\(CI\)\s+investigations\.\s*(\\n|\n)+\s*-At\s+any\s+time,\s+the\s+USG\s+may\s+inspect\s+and\s+seize\s+data\s+stored\s+on\s+this\s+IS\.\s*(\\n|\n)+\s*-Communications\s+using,\s+or\s+data\s+stored\s+on,\s+this\s+IS\s+are\s+not\s+private,\s+are\s+subject\s+to\s+routine\s+monitoring,\s+interception,\s+and\s+search,\s+and\s+may\s+be\s+disclosed\s+or\s+used\s+for\s+any\s+USG-authorized\s+purpose\.\s*(\\n|\n)+\s*-This\s+IS\s+includes\s+security\s+measures\s+\(e\.g\.,\s+authentication\s+and\s+access\s+controls\)\s+to\s+protect\s+USG\s+interests--not\s+for\s+your\s+personal\s+benefit\s+or\s+privacy\.\s*(\\n|\n)+\s*-Notwithstanding\s+the\s+above,\s+using\s+this\s+IS\s+does\s+not\s+constitute\s+consent\s+to\s+PM,\s+LE\s+or\s+CI\s+investigative\s+searching\s+or\s+monitoring\s+of\s+the\s+content\s+of\s+privileged\s+communications,\s+or\s+work\s+product,\s+related\s+to\s+personal\s+representation\s+or\s+services\s+by\s+attorneys,\s+psychotherapists,\s+or\s+clergy,\s+and\s+their\s+assistants\.\s+Such\s+communications\s+and\s+work\s+product\s+are\s+private\s+and\s+confidential\.\s+See\s+User\s+Agreement\s+for\s+details\.(\\n|\s)*$
```

It doesn't allow the shortened version of the banner, but our regex allows it.

**Comment by matejak** [comment]
This PR can't resolve the issue #4387, as it only touches the rule description.

Concerning @jan-cerny's comments, if the PR is merged, the new description of the rule would not be in sync with the rule's OVAL. Therefore, I am also not in favor of merging this. The OVAL part of the problem can get forgotten easily, and we would have false negatives :-(

**Comment by shawndwells** [comment]
Updated regex in the variable used for OVAL matches.

Used regex101.com with the following text to help test:
`````
You are accessing a U.S. Government (USG) Information System (IS) that is provided for USG-authorized use only.

By using this IS (which includes any device attached to this IS), you consent to the following conditions:

-The USG routinely intercepts and monitors communications on this IS for purposes including, but not limited to, penetration testing, COMSEC monitoring, network operations and defense, personnel misconduct (PM), law enforcement (LE), and counterintelligence (CI) investigations.

-At any time, the USG may inspect and seize data stored on this IS.

-Communications using, or data stored on, this IS are not private, are subject to routine monitoring, interception, and search, and may be disclosed or used for any USG-authorized purpose.

-This IS includes security measures (e.g., authentication and access controls) to protect USG interests--not for your personal benefit or privacy.

-Notwithstanding the above, using this IS does not constitute consent to PM, LE or CI investigative searching or monitoring of the content of privileged communications, or work product, related to personal representation or services by attorneys, psychotherapists, or clergy, and their assistants. Such communications and work product are private and confidential. See User Agreement for details.

I've read & consent to the terms in IS users agreem't
`````

**Comment by ggbecker** [comment]
It looks like the remediation is outdated for this new regex. When I run scan+remediate on `banner_etc_issue`:
```
oscap xccdf eval --remediate --profile xccdf_org.ssgproject.content_profile_stig --rule xccdf_org.ssgproject.content_rule_banner_etc_issue ssg-rhel7-ds.xml
```
It fails to remediate both `dconf_gnome_login_banner_text` and `banner_etc_issue` rules using RHEL7 STIG profile. 

When I manually paste the expected banner text to `/etc/issue`, from the comment above, the OVAL check passes.

Files which need to be updated:
https://github.com/ComplianceAsCode/content/blob/affddaa049b94efed98aed83d02d49e4f6732a3e/linux_os/guide/system/accounts/accounts-banners/gui_login_banner/dconf_gnome_login_banner_text/bash/shared.sh#L7

https://github.com/ComplianceAsCode/content/blob/affddaa049b94efed98aed83d02d49e4f6732a3e/linux_os/guide/system/accounts/accounts-banners/banner_etc_issue/bash/shared.sh#L6

**Comment by shawndwells** [comment]
Don't have time for this anymore - someone else can pickup if interested!

**Pull request closed without merging**